types/src/account_config/events/new_block.rs

```diff
@@ -141,3 +141,12 @@ impl MoveStructType for BlockResource {
 }
 
 impl MoveResource for BlockResource {}
+
+/// (keep this doc in sync with the `struct BlockLimitReachedEvent` in Move.)
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct BlockLimitReachedEvent {
```

Comment:
Should we also emit a fee receipt (exe / io gas cost + output size) when hitting the limit, or even for every block. Maybe current / future gas mechanism can leverage this information?

Reply:
not sure we want to add events to non-StateCheckpoint transaction. it is going to be very odd for user transaction that ends a block to have an event emitted (that it wasn't charged for)

maybe I can add to non-epoch-ending blocks

but, will add more details to this event irrespectively

Reply:
Depending on whether such information is useful for the gas mechanism, but I guess those two flags are sufficient already.